### PR TITLE
Search: make matches- prefix match narrower terms

### PR DESCRIPTION
**Pull request by olovy**
Before it only worked on date fields. Now it will match all `narrower`
(really `@reverse/broader`) things on non-date fields.

i.e. `?matches-instanceOf.genreForm.@id=https://id.kb.se/term/saogf/Fantasy`
will find both "Fantasy" and "Urban fantasy"

`Ranges` and `RangeParameterPrefix` should probably be renamed at some
point since they don't deal only with ranges anymore.

`matches-` still doesn't work together with `or-`, we might want to fix
this at some point.

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/search/Ranges.groovy [hunk 1]
```diff
@@ -132,4 +142,18 @@ class Ranges {
...
+    private class OrNarrower implements Query {
+        String value
+
+        OrNarrower(String value) {
+            this.value = value
+        }
+        
+        @Override
+        Map toQuery() {
+            def values = [value] + whelk.relations.followReverseBroader(value).collect() // default max size in ES 65,536...
```
Comment (olovy) [inline whelk-core/src/main/groovy/whelk/search/Ranges.groovy:155]: Note to self: Limit it instead

**Review by niklasl — APPROVED** [review]
LGTM! See my added form for the query results (and reasoning of).

**Comment by niklasl** [comment]
Added a form for the search result path in 10df18429608b76a5cbe0510a26aa4d5ea0feef3. This uses the new property defined in https://github.com/libris/definitions/pull/359.

**Comment by olovy** [comment]
> Added a form for the search result path in [10df184](https://github.com/libris/librisxl/commit/10df18429608b76a5cbe0510a26aa4d5ea0feef3). This uses the new property defined in [libris/definitions#359](https://github.com/libris/definitions/pull/359).

I buy the reasoning behind the property. LGTM!

**Pull request merged**